frontend/src/concepts/pipelines/content/artifacts/ArtifactUriLink.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import { useIsAreaAvailable, SupportedArea } from '~/concepts/areas';
+import { extractS3UriComponents } from './utils';
+
+interface ArtifactUriLinkProps {
+  uri?: string;
+}
+
+export const ArtifactUriLink: React.FC<ArtifactUriLinkProps> = ({ uri }) => {
+  const { namespace } = usePipelinesAPI();
+  const isS3EndpointAvailable = useIsAreaAvailable(SupportedArea.S3_ENDPOINT).status;
+
+  const url = React.useMemo(() => {
+    if (!uri || !isS3EndpointAvailable) {
+      return;
+    }
+
+    // Check if the uri starts with http or https return it as is
+    if (uri.startsWith('http://') || uri.startsWith('https://')) {
+      return uri;
+    }
+
+    // Otherwise check if the uri is s3
+    // If it is not s3, return undefined as we only support fetching from s3 buckets
+    const uriComponents = extractS3UriComponents(uri);
+    if (!uriComponents) {
+      return;
+    }
+
+    const { bucket, path } = uriComponents;
+
+    return `/api/storage/${namespace}/${bucket}?key=${encodeURIComponent(path)}`;
```

Comment:
Codecov says you don't have a test for this -- can you make sure you have a mocked cypress test for this?

Reply:
there is a jest test for it. it is hard to create a cypress test becuase this data is from mlmd